src/lib/bridge/handlers/fetch_recommended_price.ts

```
@@ -0,0 +1,97 @@
+import {SimpleHandler} from './main';
+import {RequestType} from './types';
+import {environment} from '../../../environment';
+import MessageSender = chrome.runtime.MessageSender;
+
+export interface CSFloatListing {
+    id: string;
+    created_at: string;
+    type: 'buy_now' | 'auction';
+    price: number; // in cents
+    state: string;
+    item: {
+        market_hash_name: string;
+    };
+}
+
+export interface FetchRecommendedPriceRequest {
+    market_hash_name: string;
+}
+
+export interface FetchRecommendedPriceResponse {
+    price: number;
+}
+
+const MIN_LISTINGS_FOR_AVERAGE = 3;
+const MAX_LISTINGS_TO_FETCH = 50;
+
+async function fetchRecommendedPrice(
+    req: FetchRecommendedPriceRequest,
+    sender: MessageSender
+): Promise<FetchRecommendedPriceResponse> {
+    // Build query parameters
+    const params: Record<string, string> = {
+        market_hash_name: req.market_hash_name,
+        limit: MAX_LISTINGS_TO_FETCH.toString(),
+        sort_by: 'lowest_price',
+        type: 'buy_now',
+    };
+
+    const url = `${environment.csfloat_base_api_url}/v1/listings?${new URLSearchParams(params)}`;
+    console.log('[FetchRecommendedPrice] Request URL:', url);
+    console.log('[FetchRecommendedPrice] Request params:', params);
+
+    try {
+        const response = await fetch(url, {
+            credentials: 'include',
+        });
+
+        if (!response.ok) {
+            const text = await response.text();
+            console.error('[FetchRecommendedPrice] API error:', response.status, text);
```

Prefer to bubble up to the caller rather than `console.error` here.

---

If using a cache, do we still want to bubble up the exception? users should still be able to list without a recommended price

---

I think we should bubble up an exception if we can't give a reference price -- users are prone to going into "autopilot" when listing items. If there is something unexpected, I'd rather that they _can't_ list an item than list it for the wrong price.